### PR TITLE
Check SPDX header in Python files

### PR DESCRIPTION
**Pull request by mwperina**
Check if Python files contains valid SPDX header with approved license

Signed-off-by: Martin Perina <mperina@redhat.com>

### CODE REVIEW

File: build-scripts/spdx-header-chech.sh [hunk 1]
```diff
@@ -12,13 +12,14 @@ APPROVED_LICENSES=""
...
 APPROVED_LICENSES="${APPROVED_LICENSES} LGPL-2.1-or-later"
+# Used in generated files in pyhirte
+APPROVED_LICENSES="${APPROVED_LICENSES} LGPL-3.0-or-later"
 
 result=0
 found_files="
     $(find -type f -regex ${CHECKED_FILES} \
         -not -path './builddir/*' \
-        -not -path './subprojects/*' \
-        -not -path './doc/arch/*')
```
Comment (engelmi) [inline build-scripts/spdx-header-chech.sh:21]: Do the docs really need the SPDX header?

**Review by engelmi — APPROVED** [review]
LGTM
Only have a small question about the header in the docs.

File: doc/api-examples/enable-unit.py [hunk 2]
```diff
@@ -1,4 +1,5 @@
 #!/usr/bin/python3
+# SPDX-License-Identifier: GPL-2.0-or-later
```
Comment (pypingou) [inline doc/api-examples/enable-unit.py]: I think we should make the example code be of a more relax license - MIT or BSD or even CC0 could be considered

Comment (mwperina) [reply doc/api-examples/enable-unit.py]: According to our license agreement those files are already published under GPL-2.0-or-later: https://github.com/containers/bluechi/tree/main/LICENSES#main-license

So this change just make the license more visible and easier to check.

If we would need to change the license of those files, then we would probably need a consent from all person contributed to those files, right?

File: build-scripts/spdx-header-chech.sh (hunk 1 above)
Comment (mwperina) [reply build-scripts/spdx-header-chech.sh:21]: Originally there were some source code files under `doc/arch`, which we wanted to omit from header check. But this directory doesn't exist anymore, so I removed it

File: doc/api-examples/enable-unit.py (hunk 2 above)
Comment (pypingou) [reply doc/api-examples/enable-unit.py]: fair, although, I would recommend we make these code snippet available under a different license

Comment (rhatdan) [reply doc/api-examples/enable-unit.py]: Yes they should be dropped down to Apache or something like it, so users could modify the script and not have to redistribute it.

Comment (mwperina) [reply doc/api-examples/enable-unit.py]: Fixed by https://github.com/containers/bluechi/pull/476 and https://github.com/containers/bluechi/pull/478

**Pull request merged**